implementation/src/main/java/io/smallrye/mutiny/operators/multi/builders/BufferItemMultiEmitter.java

```diff
@@ -83,21 +86,20 @@ void drain() {
         }
 
         int missed = 1;
-        final Queue<T> q = queue;
 
         do {
-            long r = requested.get();
-            long e = 0L;
+            long pending = requested.get();
```

Comment:
Separated "problem": one of the reason why the JDK source code for concurrent utils is very ugly ( i gently calling it "c-style") is due to an effect of the Java Memory Model:
if you have fields (eg `queue` on `queue.clear();`)  they will be read again and again (null checking, maybe implicitely, but more importantly won't benefit from being "stable") from the java, because of any surrounding volatile load and/or other atomic operations which requires mid-fields-load to happen again, to "acquire" them.
A solution would be to hoist in method-local `final` variable the fields in the hot loops; this will help the JIT (and native image)

see more info about it here -> https://psy-lob-saw.blogspot.com/2014/08/the-volatile-read-suprise.html